tensorboard/webapp/metrics/views/card_renderer/scalar_card_data_table.ts

```diff
@@ -64,29 +64,41 @@ export class ScalarCardDataTable {
     startPointIndex: number,
     endPointIndex: number,
     smoothed: boolean = false
-  ): number {
+  ): ScalarCardPoint {
     let minValue = this.maybeSmoothedValue(points[startPointIndex], smoothed);
+    let minValuePoint = points[startPointIndex];
     for (let i = startPointIndex; i <= endPointIndex; i++) {
       if (minValue > this.maybeSmoothedValue(points[i], smoothed)) {
         minValue = this.maybeSmoothedValue(points[i], smoothed);
+        minValuePoint = points[i];
       }
     }
-    return minValue;
+    return minValuePoint;
   }
 
   getMaxValueInRange(
     points: ScalarCardPoint[],
     startPointIndex: number,
     endPointIndex: number,
     smoothed: boolean = false
-  ): number {
+  ): ScalarCardPoint {
     let maxValue = this.maybeSmoothedValue(points[startPointIndex], smoothed);
+    let maxPoint = points[startPointIndex];
     for (let i = startPointIndex; i <= endPointIndex; i++) {
       if (maxValue < this.maybeSmoothedValue(points[i], smoothed)) {
         maxValue = this.maybeSmoothedValue(points[i], smoothed);
+        maxPoint = points[i];
       }
     }
-    return maxValue;
+    return maxPoint;
+  }
+
+  getMean(points: ScalarCardPoint[]) {
```

Comment:
Do you need to pass startPointIndex and endPointIndex to calculate the mean for the range? Or is points already limited to the range?